### ISSUE TITLE
QG 4 checks Release 3.2

### ISSUE THREAD

**Issue by SebastianBezold** [eclipse-tractusx/digital-product-pass#108]
## QG checks
Please keep this issue open until QG 3.2 is concluded and will be managed by the Issue Creator!
We will inform you about finding and proposals in separated issues, this issue here is for the Overview of the Checks!

Product Name: Digital Product Pass
Helm Chart Version:   
App Version:   
QG5 Approval: yes/no 

### Check of Tractus-X Release Guidelines

This QG x Check is depending on the mandatory information from our current [Release Guidelines](https://eclipse-tractusx.github.io/docs/release).

#### TRG 1 Documentation
- [x] [TRG 1.01](https://eclipse-tractusx.github.io/docs/release/trg-1/trg-1-1) appropriate `README.md` 
- [x] [TRG 1.02](https://eclipse-tractusx.github.io/docs/release/trg-1/trg-1-2) appropriate `INSTALL.md`
- [x] [TRG 1.03](https://eclipse-tractusx.github.io/docs/release/trg-1/trg-1-3) appropriate `CHANGELOG.md`
#### TRG 2 Git
- [x] [TRG 2.01](https://eclipse-tractusx.github.io/docs/release/trg-2/trg-2-1) default branch is named `main`
- [x] [TRG 2.03](https://eclipse-tractusx.github.io/docs/release/trg-2/trg-2-3) repository structure
  <details>
  <summary>Checks within TRG 2.03</summary>

  - [x] TRG 2.03 `/docs` directory contains detailed product related documentation for the Tractus-X product 
  - [x] TRG 2.03 `/charts` directory contains the Helm chart for the Tractus-X product IF available
  - [x] TRG 2.03 `AUTHORS.md` file (optional) (TRG 2.03)
  - [x] TRG 2.03 `CODE_OF_CONDUCT.md` file (TRG 2.03)
  - [x] TRG 2.03 `CONTRIBUTING.md` file (TRG 2.03)
  - [x] TRG 2.03 `DEPENDENCIES` file(s) with up to date content (Dash tool generated) (TRG 2.03)
  - [x] TRG 2.03 `LICENSE` file (TRG 2.03)
  - [x] TRG 2.03 `NOTICE.md` file (TRG 2.03)
  - [x] TRG 2.03 `SECURITY.md` file (TRG 2.03)
  </details>

- [x] [TRG 2.04](https://eclipse-tractusx.github.io/docs/release/trg-2/trg-2-4) Leading product repository
  <details>
  <summary>Checks within TRG 2.04</summary>
  
  - [x] TRG 2.04 repository name must be _productname_ without prefix or suffix
  - [x] TRG 2.04 should contain the release
  - [x] TRG 2.04 references/urls to the product's other repositories
  - [x] TRG 2.04 might contain product helm chart(s)
  - [x] TRG 2.04 README.md: contains the urls for the underlying applications
  </details>

- [x] [TRG 2.05](https://eclipse-tractusx.github.io/docs/release/trg-2/trg-2-5) `.tractusx` metafile in a proper format
#### TRG 3 Kubernetes
- [x] [TRG 3.02](https://eclipse-tractusx.github.io/docs/release/trg-3/trg-3-2) PersistentVolume and PersistentVolumeClaim is used when needed
#### TRG 4 Container
- [x] [TRG 4.01](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-1) [semantic versioning](https://semver.org/) and tagging 
- [x] [TRG 4.02](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-2) top level `README.md` file, that contains information about the used base image 
- [x] [TRG 4.03](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-3) Image has `USER` command and Non Root Container
  <details>
  <summary>Checks within TRG 4.03</summary>
  
  - [x] TRG 4.03 `deployment.yaml` has `runAsUser` and `allowPrivilegeEscalation: false` properly set
  </details>

- [x] [TRG 4.05](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-05) released image must be place `DockerHub` as mandatory container registry; remove `GHCR` references 
- [x] [TRG 4.06](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-06) Notice File for `DockerHub` has all necessary information
  <details>
  <summary>Checks within TRG 4.06</summary>

  - [x] TRG 4.06 Link to the source of your base image (Container registry and GitHub if available)
  - [x] TRG 4.06 Link to your product image on `DockerHub`
  - [x] TRG 4.06 Link to your repository on `GitHub`
  - [x] TRG 4.06 Direct link to the Dockerfile used to build your image
  - [x] TRG 4.06 Link to LICENCE file in your repo as `Project License` (make clear, that this is the PROJECT licence, not an image license
  </details>

#### TRG 5 Helm
- [x] [TRG 5.01](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-01) Helm chart must be released
  <details>
  <summary>Checks within TRG 5.01</summary>

  - [x] TRG 5.01 appropriate semantic versioning for `version` and `appVersion` has to be used in `Chart.yaml`
  - [x] TRG 5.01 must not contain any environment specific `values-xyz.yaml`
  - [x] TRG 5.01 `values.yaml` file must contain proper default values/placeholders
  - [x] TRG 5.01 No hostname provided for ingress
  - [x] TRG 5.01 Ingress is disabled
  - [x] TRG 5.01 No references to any secret engine service (e.g.: Hashicorp Vault)
  - [x] TRG 5.01 Dependencies should be prefixed with the nameOverride and/or fullnameOverride properties
  - [x] TRG 5.01 Image tag is set to the `Chart.yaml` `appVersion` property
  - [x] TRG 5.01 must be deployable to any environment without overwriting default values with a simple helm install command
  - [x] TRG 5.01 dependencies have to be declared in Chart.yaml NOT requirements.yml
  </details>
  
- [x] [TRG 5.02](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-02) Helm chart location in `/charts` directory and correct structure
  <details>
  <summary>Checks within TRG 5.02</summary>
  
  - [x] TRG 5.02 each file must contain the [Apache 2.0 Licence](https://github.com/catenax-ng/foss-example/blob/main/general/LICENSE)
  - [x] TRG 5.02 latest tag is not used in helm chart be default

  ``` markdown
  charts/ 
      chartNameA/
        Chart.yaml
        ... 
      chartNameB/
        Chart.yaml
        ...
  AUTHORS.md 
  DEPENDENCIES.md 
  LICENCE 
  README.md 
  ```
  </details>

- [x] [TRG 5.04](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-04) CPU and memory limits and requests are properly set
- [x] [TRG 5.06](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-06) application must be configurable through the Helm chart 
- [x] [TRG 5.07](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-07) dependencies are present in the `Chart.yaml` they are properly configured
- [x] [TRG 5.08](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-08) a product has a single deployable helm chart that contains all components 
  <details>
  <summary>Checks within TRG 5.08</summary>

  - [x] TRG 5.08 name of the Chart should be just the product-name without prefix or suffix
  - [x] TRG 5.08 values file should contain all available variables (even from subcharts) with default values and comments about what they do
  - [x] TRG 5.08 helm install command should successfully install the chart to any supported Kubernetes version cluster (without overwriting default values)
  - [x] TRG 5.08 helm test runs without errors
  </details>
  
- [x] [TRG 5.09](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-09) Helm Test running properly
  <details>
  <summary>Checks within TRG 5.09</summary>

  - [x] TRG 5.09 A GitHub action exist which builds or uses the helm chart which gets released
  - [x] TRG 5.09 The GitHub action can be triggered manually through Github WebUI manually running a workflow
  - [x] TRG 5.09 Helm test verifies that the application is up and running
  </details>
- [x] [TRG 5.10](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-10) Products need to support 3 versions at a time
  <details>
  <summary>Checks within TRG 5.10</summary>

  - [x] TRG 5.10 latest (K8s version 1.25)
  - [x] TRG 5.10 latest - 1 (K8s version 1.24)
  - [x] TRG 5.10 latest - 2  (K8s version 1.23)
  </details>
- [x] [TRG 5.11](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-11) Upgradeability PRERELEASE
  <details>
  <summary>Checks within TRG 5.11</summary>

  - [x] TRG 5.11 Based on the Helm test workflow, you must provide a GitHub action which takes the latest released helm chart, does an installation of it and then execute the upgrade to the current / new version.
  </details>
  
#### TRG 6 Released Helm Chart 
- [x] [TRG 6.01](https://eclipse-tractusx.github.io/docs/release/trg-6/trg-6-1) Released Helm Chart 
#### TRG 7 Open Source Governance
- [x] [TRG 7.01](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-01) Legal Documentation
- [x] [TRG 7.02](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) License and copyright header  
- [x] [TRG 7.03](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-03) IP checks for project content 
- [x] [TRG 7.04](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04) IP checks for 3rd party content
  <details>
  <summary>Checks within TRG 7.04</summary>

  - [x] TRG 7.04 DEPENDENCIES file is up-to-date and reflects the current use of the 3rd party content
  - [x] TRG 7.04 all libraries listed there should have the status "approved"
  - [x] TRG 7.04 no libraries with status "rejected"
  - [x] TRG 7.04 for libraries with status "restricted", the according IP issues must be present (issue number in the source column)
  </details>
- [x] [TRG 7.05](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-05) Legal information for distributions
- [x] [TRG 7.06](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-06) Legal information for end user content
- [ ] [TRG 7.07](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-07) Legal notice for documentation

**Comment by SebastianBezold** [comment]
Hi @matbmoser and @saudkhan116,
i don't have any info from release management regarding the digital-product-pass version, that should be included in the 3.2 releaese. Is it the 1.0.0-alpha? Will you release it as 1.0.0 if the release guideline checks are passed?

**Comment by matbmoser** [comment]
Hi @SebastianBezold,  We will release a release candidate for 1.0.0 today. It contain all the Quality Gates requirements as documented in GitHub. The release v1.0.0 does not contains the complete requirements.  Because of this we did not contacted with you yet.

**Comment by matbmoser** [comment]
@SebastianBezold the released was not created before in the feature freaze because of some architecture requirements that were pending to be tested.

**Comment by matbmoser** [comment]
https://github.com/eclipse-tractusx/digital-product-pass/issues/33 We have this already created and checked here 👍🏻

**Comment by SebastianBezold** [comment]
Hi @matbmoser,
ok thanks for the input. I created the issue, since we where triggered by Release management. 
So [v1.0.0-rc1](https://github.com/eclipse-tractusx/digital-product-pass/releases/tag/v1.0.0-rc1) is the "ready" version?

**Comment by matbmoser** [comment]
After the E2E Tests worked and all the security requirements are completed a version v1.0.0 will be created, and will be the one ready for QG4. From a this perspective this version v1.0.0-rc2 should contain all the TRG requirements, and all that is in this version will be included in the v1.0.0

**Comment by matbmoser** [comment]
@SebastianBezold  v1.0.0 is released after passing our internal E2E tests! https://github.com/eclipse-tractusx/digital-product-pass/releases/tag/v1.0.0

**Comment by SebastianBezold** [comment]
Hi @matbmoser,
I think I don't want to create an issue for it, but maybe for consideration:
There is an `avp` key in your [Charts values](https://github.com/eclipse-tractusx/digital-product-pass/blob/v1.0.0/charts/digital-product-pass/values.yaml#L92). Not sure, if in DPP context, AVP does make sense, but if you use it as `Argo CD Vault Plugin` I would rename it. 
This is too heavily tied to a specific deployment tooling

**Comment by matbmoser** [comment]
> Hi @matbmoser, I think I don't want to create an issue for it, but maybe for consideration: There is an `avp` key in your [Charts values](https://github.com/eclipse-tractusx/digital-product-pass/blob/v1.0.0/charts/digital-product-pass/values.yaml#L92). Not sure, if in DPP context, AVP does make sense, but if you use it as `Argo CD Vault Plugin` I would rename it. This is too heavily tied to a specific deployment tooling

Ok, thank you for making this visible, we will refactor it for the next release.

**Comment by SebastianBezold** [comment]
Hi @matbmoser, I linked the issue #122 here, because I don't really understand the Chart testing workflow here. Could you or anyone else from the team please take a look and clarify?

**Comment by matbmoser** [comment]
Ok we fixed the problem now everything is up and running in the release v1.0.0

**Comment by SebastianBezold** [comment]
Current status: I want to test the changes made in `v1.0.0-rc4`, but the Docker images don't seem to be release yet

**Comment by saudkhan116** [comment]
Hi @SebastianBezold, 
as far as I know, the license dash tool is down and we are not able to publish docker images for the latest release version v1.0.0. Yesterday, the release version v1.0.0 was updated.

**Comment by SebastianBezold** [comment]
@saudkhan116,
did you dependencies change for the -rc4 changes, or what do you need dash for in regards to docker build?
Does the update of `v1.0.0` include the changes done for -rc4?

**Comment by saudkhan116** [comment]
We didn't change any dependencies, there was a hotfix #125 yesterday that got merged into the `v1.0.0` . Due to this, we need to run the dash tool to scan the dependencies check again and afterwards publish the images.

**Comment by saudkhan116** [comment]
@SebastianBezold  the new docker images are now published here:

- https://hub.docker.com/r/tractusx/digital-product-pass-backend
- https://hub.docker.com/r/tractusx/digital-product-pass-frontend

**Comment by SebastianBezold** [comment]
Hi @saudkhan116 and @matbmoser,

almost all the checks are passed. Thanks for the fixes!
There is just one last thing, which is legal notice in documentation [TRG 7.07](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-07/#documentation). I linked it to the specific section.
TL;DR on what is missing.
The docs, that you have in the `/docs` folder should also contain a notice to the License they are put under (In your case most likely `Apache-2.0`. This section should be readable by the end user. This means no comment like in the source code files but plain markdown. 
It would be enough, if the "Entrypoint" to the documentation does contain that notice. In case you have different types of docs (i.e. Arc42, User Guide, Admin Guide, etc.) then every "Entrypoint" to this docs should have the Notice.

If you add these Notices, then I'll conclude the QG checks, you also do not need to build a new version for that change. `main` is fine here

**Comment by SebastianBezold** [comment]
Sorry small update @matbmoser and @saudkhan116, I was wrong here.
Since the files are rendered separately when viewing, we actually need the `## Notice` section in every file

**Comment by saudkhan116** [comment]
> Sorry small update @matbmoser and @saudkhan116, I was wrong here. Since the files are rendered separately when viewing, we actually need the `## Notice` section in every file

Hi @SebastianBezold, thank you for your feedback. We will have look into the [TRG 7.07](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-07/#documentation)

**Comment by matbmoser** [comment]
Reopened the issue  #60 which was closed due to not enough clarification.

**Comment by saudkhan116** [comment]
Hi @SebastianBezold, the requested TRG requirement is implemented in PR #127. Please have a review.
Thanks!

**Comment by saudkhan116** [comment]
Thank you very much @SebastianBezold 👍💯

**Comment by matbmoser** [comment]
> Hi @saudkhan116, thanks a lot! I guess you can close #60 with the merge of #127 as well. CC-BY-4.0 is definitely needed for KIT docs on the adoption view. For all other products, you can choose to put more docs under CC-BY, but it's not mandatory.
> 
> With this done, the QG 4 checks from my side are done :) congrats 👍 🚀

Thank you Sebastian for the review! 🥇
